### PR TITLE
Support custom headers

### PR DESCRIPTION
**Pull request by ugurakkurt**
Fixes https://github.com/rgommezz/react-native-offline/issues/274

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,7 @@ type Props = {
     pingOnlyIfOffline?: boolean = false,
     pingInBackground?: boolean = false,
     httpMethod?: HTTPMethod = 'HEAD',
+    customHeaders?: {} = {},
```
Comment (rgommezz) [inline README.md:123]: Type should be:

```js
type HTTPHeaders = {
	[key: string]: string
}
...

customHeaders?: HTTPHeaders = {}
```

You can add the new type [here](https://github.com/rgommezz/react-native-offline/blob/master/src/types.ts) and export it accordingly.

File: src/types.ts [hunk 2]
```diff
@@ -40,6 +40,7 @@ export type ConnectivityArgs = {
   pingOnlyIfOffline: boolean;
   pingInBackground: boolean;
   httpMethod: HTTPMethod;
+  customHeaders: {};
```
Comment (rgommezz) [inline src/types.ts:43]: Please change this to `HTTPHeaders` type as per my comment above

File: src/utils/checkInternetConnection.ts [hunk 3]
```diff
@@ -20,13 +21,15 @@ export default async function checkInternetConnection(
   timeout: number = DEFAULT_TIMEOUT,
   shouldPing = true,
   method: HTTPMethod = DEFAULT_HTTP_METHOD,
+  customHeaders: {} = DEFAULT_CUSTOM_HEADERS,
```
Comment (rgommezz) [inline src/utils/checkInternetConnection.ts:24]: Change type to `HTTPHeaders`

File: src/utils/checkInternetAccess.ts [hunk 4]
```diff
@@ -3,19 +3,22 @@ import {
...
   timeout: number;
   method?: HTTPMethod;
+  customHeaders: {};
 };
 
 const DEFAULT_ARGUMENTS: Arguments = {
   timeout: DEFAULT_TIMEOUT,
   url: DEFAULT_PING_SERVER_URL,
   method: DEFAULT_HTTP_METHOD,
+  customHeaders: {},
```
Comment (rgommezz) [inline src/utils/checkInternetAccess.ts:21]: Use `DEFAULT_CUSTOM_HEADERS`

File: src/utils/makeHttpRequest.ts [hunk 5]
```diff
@@ -16,6 +16,7 @@ type Options = {
     | 'onload/5xx'
     | 'onerror'
     | 'ontimeout';
+  customHeaders?: {};
```
Comment (rgommezz) [inline src/utils/makeHttpRequest.ts:19]: Change type to `HTTPHeaders`

File: src/utils/makeHttpRequest.ts [hunk 6]
```diff
@@ -42,12 +44,14 @@ const DEFAULT_OPTIONS: Options = {
   method: DEFAULT_HTTP_METHOD,
   url: DEFAULT_PING_SERVER_URL,
   timeout: DEFAULT_TIMEOUT,
+  customHeaders: {},
```
Comment (rgommezz) [inline src/utils/makeHttpRequest.ts:47]: Use `DEFAULT_CUSTOM_HEADERS`

File: src/utils/makeHttpRequest.ts [hunk 7]
```diff
@@ -42,12 +44,14 @@ const DEFAULT_OPTIONS: Options = {
...
 };
 export default function makeHttpRequest(args?: Options) {
   const {
     method = DEFAULT_HTTP_METHOD,
     url = DEFAULT_PING_SERVER_URL,
     timeout = DEFAULT_TIMEOUT,
+    customHeaders = {},
```
Comment (rgommezz) [inline src/utils/makeHttpRequest.ts:54]: Use `DEFAULT_CUSTOM_HEADERS`

File: test/checkInternetAccess.test.ts [hunk 8]
```diff
@@ -44,16 +48,40 @@ describe('checkInternetAccess', () => {
...
+      url,
+      timeout,
+      customHeaders: customHeadersToAdd,
+    });
+    expect(makeHttpRequest).toHaveBeenCalledWith({
+      url,
+      timeout,
+      method: DEFAULT_HTTP_METHOD,
+      customHeaders: customHeadersToAdd,
+
```
Comment (rgommezz) [inline test/checkInternetAccess.test.ts:83]: Nit: remove this empty line

File: test/makeHttpRequest.test.ts [hunk 9]
```diff
@@ -117,6 +118,11 @@ describe('makeHttpRequest', () => {
     expect(mockSend).toHaveBeenCalledWith(null);
   });
 
+  it('accepts custom headers', () => {
+    makeHttpRequest({ ...params, customHeaders: { foo: 'bar' } });
+    expect(mockSetRequestHeader).toHaveBeenCalledTimes(4);
```
Comment (rgommezz) [inline test/makeHttpRequest.test.ts:123]: Nit: a better expectation would be:

```js
expect(mockSetRequestHeader).toHaveBeenNthCalledWith(4, 'foo', 'bar');
```

**Comment by ugurakkurt** [comment]
Thank you for the review :) @rgommezz

**Comment by rgommezz** [comment]
@all-contributors please add @ugurakkurt for code and docs

**Pull request merged**